src/bus/webSocketConnection.spec.js

```diff
@@ -442,4 +441,527 @@ describe('Bus/WebSocketConnection', function() {
+        return promise.catch(function() {
+          expect(ws._jsonRpcId).to.equal(jsonRpcId);
+        });
+      });
+
+      it('rejects the promise', function() {
+        return expect(promise).to.be.rejectedWith(
+          'WebSocket connection not open'
+        );
+      });
+    });
+
+    context('when there is not a service id sent', function() {
+      let channel;
+      let expectedOrganization;
+      let jsonRpcId;
+      let message;
+      let promise;
+      let send;
+      let serviceId;
+      let ws;
+
+      beforeEach(function() {
+        channel = faker.random.word();
+        expectedOrganization = fixture.build('organization');
+        message = {
+          example: 1
+        };
+        send = this.sandbox.spy(expectedWebSocket, 'send');
+        serviceId = null;
+
+        ws = new WebSocketConnection(
+          expectedWebSocket,
+          expectedOrganization.id
+        );
+
+        jsonRpcId = ws._jsonRpcId;
+
+        promise = ws.publish(serviceId, channel, message);
+      });
+
+      it('does not send a message to the message bus', function() {
+        return promise.catch(function() {
+          expect(send).to.not.be.called;
+        });
+      });
+
+      it('does not increment the jsonRpcId', function() {
+        return promise.catch(function() {
+          expect(ws._jsonRpcId).to.equal(jsonRpcId);
+        });
+      });
+
+      it('rejects the promise', function() {
+        return expect(promise).to.be.rejectedWith(
+          'A service client id is required for publishing'
+        );
+      });
+    });
+
+    context('when there is not a channel sent', function() {
+      let channel;
+      let expectedOrganization;
+      let jsonRpcId;
+      let message;
+      let promise;
+      let send;
+      let serviceId;
+      let ws;
+
+      beforeEach(function() {
+        channel = null;
+        expectedOrganization = fixture.build('organization');
+        message = {
+          example: 1
+        };
+        send = this.sandbox.spy(expectedWebSocket, 'send');
+        serviceId = faker.random.uuid();
+
+        ws = new WebSocketConnection(
+          expectedWebSocket,
+          expectedOrganization.id
+        );
+
+        jsonRpcId = ws._jsonRpcId;
+
+        promise = ws.publish(serviceId, channel, message);
+      });
+
+      it('does not send a message to the message bus', function() {
+        return promise.catch(function() {
+          expect(send).to.not.be.called;
+        });
+      });
+
+      it('does not increment the jsonRpcId', function() {
+        return promise.catch(function() {
+          expect(ws._jsonRpcId).to.equal(jsonRpcId);
+        });
+      });
+
+      it('rejects the promise', function() {
+        return expect(promise).to.be.rejectedWith(
+          'A channel is required for publishing'
+        );
+      });
+    });
+
+    context('when there is not a message sent', function() {
```

Comment:
Is there an easy way to do some code reuse on these tests that are almost all the same?